website/addons/base/views.py

```diff
@@ -283,8 +283,15 @@ def create_waterbutler_log(payload, **kwargs):
 
         node_addon.create_waterbutler_log(auth, action, metadata)
 
-    return {'status': 'success'}
+        try:
+            action = payload['action']
```

Comment:
Multiple lines in a `try` block are a red flag - you should keep the operations inside its scope as limited as possible.

Even pulling multiple keys out of a nested dict here, you run the risk of the format of the dict changing subtly in the future, raising a `KeyError` on only one of the assignments, then silently passing.

Calling `node.update_search_file` is more dangerous still - if anything inside that function or one that it ultimately calls raises a `KeyError`, for any reason, that error will get swallowed up. 

This is one of the very few (only?) use cases for Python's `try...except..else`. Consider the following:

```python
try:
    action = payload['action']
    name = payload['metadata']['name']
    path = payload['metadata']['path']
except KeyError:
    pass
else:
    # only gets called if no exception was raised
    node.update_search_file(action, node_addon, name, path)
```

Doing it this way ensures that `node.update_search_file` is called only if an exception is not raised, while keeping your call outside of the `try` block.

I'd leave the comment there in production code, too - this feature is used very infrequently and is easily misunderstood, so the reminder could prevent someone "fixing" it in the future and breaking it :)